manager/controlapi/secret.go

```diff
@@ -238,16 +238,26 @@ func (s *Server) RemoveSecret(ctx context.Context, request *api.RemoveSecretRequ
 	}
 }
 
+// ValidateSecretPayload validates the secret payload size
+func ValidateSecretPayload(data []byte) error {
```

Comment:
Moved to `api/validation/secrets.go`, I kept the prefix validation in the function name, let me know if you prefer `validation.SecretPayload`